src/client/index.tsx

```diff
@@ -16,87 +16,45 @@ __webpack_public_path__ = getPublicPath();
-import { JupyterNotebookRenderer } from './constants';
 import { CellOutput } from './render';
+import { ActivationFunction, CellInfo } from 'vscode-notebook-renderer';
 
-const notebookApi = acquireNotebookRendererApi(JupyterNotebookRenderer);
-
-notebookApi.onDidCreateOutput(renderOutput);
-
-// Copy of vscode-notebook-renderer old types as of 1.48
-// Keep these so we can support both the old interface and the new interface
-// Interface change here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/51675/files
-interface OldNotebookCellOutputMetadata {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    custom?: { [key: string]: any };
-}
-interface OldNotebookOutput {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    data: { [mimeType: string]: any };
-    metadata?: OldNotebookCellOutputMetadata;
-}
-interface OldNotebookOutputEventParams {
-    element: HTMLElement;
-    outputId: string;
-    output: OldNotebookOutput;
-    mimeType: string;
-}
+export const activate: ActivationFunction = () => ({
+    renderCell(_id, cellInfo: CellInfo) {
+        renderOutput(cellInfo);
+    }
+});
 
 /**
  * Called from renderer to render output.
  * This will be exposed as a public method on window for renderer to render output.
  */
-function renderOutput(request: NotebookOutputEventParams) {
+function renderOutput(cellInfo: CellInfo) {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const mimeString = request.mime || (request as any).mimeType;
+    const mimeString = cellInfo.mime || (cellInfo as any).mimeType;
     try {
-        console.log('request', request);
-        const output = convertVSCodeOutputToExecutResultOrDisplayData(request);
+        console.log('request', cellInfo);
+        const output = convertVSCodeOutputToExecutResultOrDisplayData(cellInfo);
         console.log(`Rendering mimeType ${mimeString}`, output);
 
-        ReactDOM.render(React.createElement(CellOutput, { mimeType: mimeString, output }, null), request.element);
+        ReactDOM.render(React.createElement(CellOutput, { mimeType: mimeString, output }, null), cellInfo.element);
     } catch (ex) {
         console.error(`Failed to render mime type ${mimeString}`, ex);
     }
 }
 
 function convertVSCodeOutputToExecutResultOrDisplayData(
-    request: NotebookOutputEventParams
+    cellInfo: CellInfo
 ): nbformat.IExecuteResult | nbformat.IDisplayData {
-    if ('mime' in request) {
-        // New API
-        return {
-            data: {
-                [request.mime]: request.value
-            },
-            metadata: request.metadata || {},
-            execution_count: null,
-            output_type: request.metadata?.outputType || 'execute_result'
-        };
-    } else {
-        // Old API
+    return {
+        data: {
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            [cellInfo.mime]: cellInfo.value as any
```

Comment:
The any types are a bit ugly here, but it's actually going to those super loose Jupyter JSONObjects and I didn't want to import phosphor and whatnot for something that didn't really give any type safety. So went with any for now, but not adverse to looking if we can make these stronger types in a follow up.